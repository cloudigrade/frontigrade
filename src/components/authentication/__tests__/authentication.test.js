import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import Authentication from '../authentication';

describe('Authentication Component', () => {
  const generateEmptyStore = obj => configureMockStore()(obj);

  it('should shallow snapshot a basic component with login', () => {
    const store = generateEmptyStore({ user: { session: { error: false, loginFailed: false, pending: true } } });
    const wrapper = shallow(
      <Authentication>
        <span className="test">lorem</span>
      </Authentication>,
      { context: { store } }
    );

    expect(wrapper).toMatchSnapshot();

    const wrapperHtml = wrapper.html();
    expect(wrapperHtml).toContain('Loading...');
    expect(wrapperHtml).not.toContain('Email address or password is incorrect');
  });

  it('should have a login error', () => {
    const store = generateEmptyStore({ user: { session: { error: true, loginFailed: true } } });
    const wrapper = shallow(
      <Authentication>
        <span className="test">lorem</span>
      </Authentication>,
      { context: { store } }
    );

    expect(wrapper.html()).toContain('Email address or password is incorrect');
  });

  it('should shallow render a basic component without login', () => {
    const store = generateEmptyStore({ user: { session: { authorized: true } } });
    const wrapper = shallow(
      <Authentication>
        <span id="test">lorem</span>
      </Authentication>,
      { context: { store } }
    );

    expect(wrapper.render()).toMatchSnapshot();
  });
});
