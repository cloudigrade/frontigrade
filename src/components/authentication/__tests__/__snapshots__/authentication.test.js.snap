// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected 1`] = `
<Authentication
  checkUser={[Function]}
  loginUser={[Function]}
  removeStoredData={[Function]}
  session={
    Object {
      "error": false,
      "loginFailed": false,
      "pending": false,
    }
  }
  storeData={[Function]}
>
  <span
    className="test"
  >
    lorem
  </span>
</Authentication>
`;

exports[`expected state 1`] = `
Object {
  "email": "",
  "emailError": "Email must be valid",
  "formTouched": true,
  "formValid": false,
  "password": "123",
  "passwordError": "",
  "remember": true,
}
`;

exports[`post-authorization 1`] = `
<span
  class="test"
>
  lorem
</span>
`;
